docs/advanced/writing-shaders.md

```diff
@@ -64,6 +64,34 @@ The layerIndex is a small integer that starts at zero and is incremented for eac
 
 In the fragment shader, multiply the fragment color with the opacity uniform.
 
+### Picking uniforms (deprecated)
+
+NOTE: Following uniforms (`renderPickingBuffer` and `selectedPickingColor`) are supported as deprecated, and will be removed in next major version. It is recommended to use `luma.gl` picking module for any picking needs. For more details refer to [`Picking`](/docs/advanced/picking.md).
+
+##### `float renderPickingBuffer`
```

Comment:
This section was added from 4.1-release.

Reply:
Add deprecation notes to "Upgrade Guide" as well?